src/views/pacbio/PacbioLibraryBatchCreate.vue

```diff
@@ -0,0 +1,302 @@
+                      @click="onPrintAction"
+                    >
+                      Print Labels
+                    </traction-button>
+                  </div>
+
+                  <div tag="article" class="mb-2 text-black text-left">
+                    <div class="flex flex-col rounded p-4">
+                      <ul
+                        v-if="printBarcodes.length"
+                        id="list-barcodes-to-print"
+                        class="space-y-2 justify-center text-center"
+                      >
+                        <li
+                          v-for="barcode in printBarcodes"
+                          :key="barcode"
+                          class="text-sm text-sp font-bold"
+                        >
+                          {{ barcode }}
+                        </li>
+                      </ul>
+                      <div
+                        v-else
+                        data-type="empty-barcodes"
+                        class="justify-center text-center text-black text-sm font-bold"
+                      >
+                        No Barcodes to Print Yet
+                      </div>
+                    </div>
+                  </div>
+                </div>
+              </div>
+            </div>
+          </div>
+        </div>
+      </div>
+    </div>
+  </DataFetcher>
+</template>
+
+<script setup>
+/**
+ * PacbioLibraryBatchCreate.vue
+ *
+ * This component is used to create a library batch from a csv file and tag set
+ * and print labels for the created libraries
+ *
+ * @module PacbioLibraryBatchCreate
+ */
+import { usePacbioRootStore } from '@/stores/pacbioRoot.js'
+import { usePrintingStore } from '@/stores/printing.js'
+import { usePacbioLibraryBatchCreateStore } from '@/stores/pacbioLibraryBatchCreate.js'
+import { computed, ref, reactive } from 'vue'
+import DataFetcher from '@/components/DataFetcher.vue'
+import { eachRecord } from '@/lib/csv/pacbio.js'
+import useAlert from '@/composables/useAlert.js'
+import usePacbioLibraryPrint from '@/composables/usePacbioLibraryPrint.js'
+
+const pacbioRootStore = usePacbioRootStore()
+const pacbioLibraryBatchCreateStore = usePacbioLibraryBatchCreateStore()
+const { printers, fetchPrinters } = usePrintingStore()
+const { showAlert } = useAlert()
+const { printLabels } = usePacbioLibraryPrint()
+
+const defaultTagSetName = 'Pacbio_96_barcode_plate_v3'
+
+const selectedPrinterName = ref('') // selected printer id
+const showCSVPreview = ref(false) // Show preview of the csv file
+const selectedTagSet = ref('') // Chosen tag set
+const csvFileInput = ref('') //Reference to the csv file input component
+/**
+ * Set the default tag set
+ *
+ */
+const setDefaultTagSet = () => {
+  const defaultTagSet = pacbioRootStore.tagSetChoices.find(
+    (option) => option.text === defaultTagSetName,
+  )
+  if (defaultTagSet) {
+    selectedTagSet.value = defaultTagSet.value
+  }
+}
+
+async function fetchData() {
```

Comment:
The page is unusable if the data fetch fails so not sure if this should always return true.

Reply:
Good point, will change it to displaying an error message in case of failure